packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -1098,4 +1101,75 @@ export class PrepareProvider extends Provider implements PreparationProvider {
       contents,
     };
   }
+
+  /**
+   *
+   * @description Method to prepare a stake operation
+   * @param params RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  stake(params: StakingParams): Promise<PreparedOperation> {
+    return this.staking(params, 'stake');
+  }
+
+  /**
+   *
+   * @description Method to prepare an unstake operation
+   * @param params RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  async unstake(params: StakingParams): Promise<PreparedOperation> {
+    return this.staking(params, 'unstake');
+  }
+
+  /**
+   *
+   * @description Method to prepare a finalize_unstake operation
+   * @param params RPCOperation object
+   * @returns a PreparedOperation object
+   */
+  async finalizeUnstake(params: StakingParams): Promise<PreparedOperation> {
+    return this.staking(params, 'finalize_unstake');
+  }
+
+  private async staking(
+    { fee, storageLimit, gasLimit, sourceAndDestination, ...rest }: StakingParams,
+    entrypoint: StakingEntrypoint
+  ): Promise<PreparedOperation> {
+    const { pkh } = await this.getKeys();
+
+    const protocolConstants = await this.context.readProvider.getProtocolConstants('head');
+    const DEFAULT_PARAMS = await this.getAccountLimits(pkh, protocolConstants);
+    const mergedEstimates = mergeLimits({ fee, storageLimit, gasLimit }, DEFAULT_PARAMS);
+
+    const op = await createStakingOperation(
```

Comment:
might want to also reference line1025 of how to prepare contractCall to see if it fits pseudo operation's need better than transaction